test/preconditioner/isai_kernels.cpp

```diff
@@ -275,43 +263,43 @@ TEST_F(Isai, HipIsaiGenerateUinverseLongIsEquivalentToRef)
 }
 
 
-TEST_F(Isai, HipIsaiGenerateAinverseLongIsEquivalentToRef)
+TEST_F(Isai, IsaiGenerateAinverseLongIsEquivalentToRef)
 {
     initialize_data(matrix_type::general, 695, 64);
     const auto num_rows = mtx->get_size()[0];
     gko::array<index_type> a1(ref, num_rows + 1);
     auto a2 = a1;
-    gko::array<index_type> da1(hip, num_rows + 1);
+    gko::array<index_type> da1(exec, num_rows + 1);
     auto da2 = da1;
 
     gko::kernels::reference::isai::generate_general_inverse(
         ref, mtx.get(), inverse.get(), a1.get_data(), a2.get_data(), false);
-    gko::kernels::hip::isai::generate_general_inverse(
-        hip, d_mtx.get(), d_inverse.get(), da1.get_data(), da2.get_data(),
+    gko::kernels::EXEC_NAMESPACE::isai::generate_general_inverse(
+        exec, d_mtx.get(), d_inverse.get(), da1.get_data(), da2.get_data(),
         false);
 
     GKO_ASSERT_MTX_EQ_SPARSITY(inverse, d_inverse);
-    GKO_ASSERT_MTX_NEAR(inverse, d_inverse, 10 * r<value_type>::value);
+    GKO_ASSERT_MTX_NEAR(inverse, d_inverse, 100 * r<value_type>::value);
```

Comment:
maybe need to be careful?

Reply:
I like to increase the error bounds with big margins to make sure we don't have any flaky tests if some compiler optimizations change the evaluation order. Do you think this is an issue here?

Reply:
I do not know actually.
Without knowing the testing problem, I can somehow accept 10 r as rounding issue. 100 r will be a little higher such that I start wondering whether there's something wrong or the wrong codes may also pass the test